packages/core/src/jobs/job-definitions/batchEvaluations/runEvaluationJob.test.ts

```diff
@@ -61,84 +42,181 @@ vi.spyOn(progressTracker, 'ProgressTracker').mockImplementation(() => ({
-        // @ts-ignore
-        Promise.resolve({
-          response: responsePromise,
+    it('increment successful counter', async () => {
+      runChainResponse = Result.ok({
+        streamType: 'object' as 'object',
+        object: { result: { result: '42', reason: 'Is always 42' } },
+        text: 'chain resolved text',
+        usage: { promptTokens: 8, completionTokens: 2, totalTokens: 10 },
+        documentLogUuid: documentLog.uuid,
+        providerLog: undefined,
+      })
+      runEvaluationSpy.mockResolvedValueOnce(
+        Result.ok({
+          stream,
+          response: new Promise((resolve) => resolve(runChainResponse)),
+          resolvedContent: 'chain resolved text',
+          errorableUuid: FAKE_ERRORABLE_UUID,
+          duration: new Promise((resolve) => resolve(1000)),
         }),
+      )
+      await runEvaluationJob(jobData)
+
+      expect(websockets.WebsocketClient.getSocket).toHaveBeenCalledTimes(1)
+      expect(incrementCompletedSpy).toHaveBeenCalledTimes(1)
+      expect(mockEmit).toHaveBeenCalledWith('evaluationStatus', {
+        workspaceId: workspace.id,
+        data: {
+          batchId: 'batch-123',
+          evaluationId: evaluation.id,
+          documentUuid,
+          completed: 1,
+          total: 1,
+        },
+      })
     })
 
-    const awaitResponseSpy = vi.fn()
-    responsePromise.then(awaitResponseSpy)
+    it('increment error counter', async () => {
+      runEvaluationSpy.mockResolvedValueOnce(
+        Result.error(
+          new ChainError({
+            code: RunErrorCodes.EvaluationRunResponseJsonFormatError,
+            message: 'malformed json response',
+          }),
+        ),
+      )
+
+      await runEvaluationJob(jobData)
+
+      expect(websockets.WebsocketClient.getSocket).toHaveBeenCalledTimes(1)
+      expect(incrementErrorsSpy).toHaveBeenCalledTimes(1)
+      expect(mockEmit).toHaveBeenCalledWith('evaluationStatus', {
+        workspaceId: workspace.id,
+        data: {
+          batchId: 'batch-123',
+          evaluationId: evaluation.id,
+          documentUuid,
+          completed: 1,
+          total: 1,
+        },
+      })
+    })
 
-    await runEvaluationJob(jobData)
-    expect(awaitResponseSpy).toHaveBeenCalled()
-  })
+    it('throws an error if runEvaluation fails', async () => {
+      runEvaluationSpy.mockRejectedValue(new Error('Some error'))
 
-  it('increment successful counter', async () => {
-    await runEvaluationJob(jobData)
-
-    expect(websockets.WebsocketClient.getSocket).toHaveBeenCalledTimes(1)
-    expect(incrementCompletedSpy).toHaveBeenCalledTimes(1)
-    expect(mockEmit).toHaveBeenCalledWith('evaluationStatus', {
-      workspaceId: 1,
-      data: {
-        batchId: 'batch-123',
-        evaluationId: 2,
-        documentUuid: 'doc-uuid',
-        completed: 1,
-        total: 1,
-      },
+      await expect(runEvaluationJob(jobData)).rejects.toThrowError(
+        new Error('Some error'),
+      )
```

Comment:
This is testing that we don't need to wait for the response. It was weird

```javascript
const { response } = runEvaluation()
await response
```

Now it will fail if 
1. There is an error before AI is called
2. There is an error in AI response

All we have to do is wait to finish and we get a Result
```javascript
const result = runEvaluation()
```